### PR TITLE
[TextServer] Use GDVIRTUAL_REQUIRED_CALL for required extensions methods, update extension docs.

### PR DESCRIPTION
**Pull request by bruvzg**
- Adds `GDVIRTUAL_REQUIRED_CALL` to required methods.
- Updates extension docs.
- Updates dummy text server to implement required methods.

### CODE REVIEW

File: servers/text/text_server_dummy.h [hunk 1]
```diff
@@ -40,9 +40,87 @@ class TextServerDummy : public TextServerExtension {
...
+	const Glyph *shaped_text_sort_logical(const RID &p_shaped) override { return nullptr; };
+	int64_t shaped_text_get_glyph_count(const RID &p_shaped) const override { return 0; }
+	Vector2i shaped_text_get_range(const RID &p_shaped) const override { return Vector2i(); }
+	int64_t shaped_text_get_trim_pos(const RID &p_shaped) const override { return -1; }
+	int64_t shaped_text_get_ellipsis_pos(const RID &p_shaped) const override { return -1; }
+	const Glyph *shaped_text_get_ellipsis_glyphs(const RID &p_shaped) const override { return nullptr; }
+	int64_t shaped_text_get_ellipsis_glyph_count(const RID &p_shaped) const override { return -1; }
+	Array shaped_text_get_objects(const RID &p_shaped) const override { return Array(); }
+	Rect2 shaped_text_get_object_rect(const RID &p_shaped, const Variant &p_key) const override { return Rect2(); }
+	Vector2i shaped_text_get_object_range(const RID &p_shaped, const Variant &p_key) const override { return Vector2i(); };
```
Comment (AThousandShips) [inline servers/text/text_server_dummy.h:116]:
```suggestion
	Vector2i shaped_text_get_object_range(const RID &p_shaped, const Variant &p_key) const override { return Vector2i(); }
```

File: servers/text/text_server_dummy.h [hunk 2]
```diff
@@ -40,9 +40,87 @@ class TextServerDummy : public TextServerExtension {
...
+	int64_t shaped_text_get_object_glyph(const RID &p_shaped, const Variant &p_key) const override { return -1; };
```
Comment (AThousandShips) [inline servers/text/text_server_dummy.h:117]:
```suggestion
	int64_t shaped_text_get_object_glyph(const RID &p_shaped, const Variant &p_key) const override { return -1; }
```

File: servers/text/text_server_dummy.h
Comment (AThousandShips) [inline servers/text/text_server_dummy.h:1]: All these should have `virtual` as well as per the general style norm

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**